harness/determined/cli/experiment.py

```diff
@@ -670,7 +672,9 @@ def kill_experiment(args: Namespace) -> None:
 
 @authentication.required
 def wait(args: Namespace) -> None:
```

Comment:
can you elaborate on what you mean? by that they don't need to "actually hydrate an experiment" and why would it mean that something is wrong?

Reply:
yeah, i thought about this. see my comment in this PR description:
"objects constructed by the user will not. this is by design, because sometimes methods on a resource class function more as utility methods for the class, and do not require fully-hydrated properties (eg. TrialReference(id=1).get_checkpoints() will not fetch the initial TrialReference properties, and just call the separate checkpoints REST API)"

Reply:
Thanks. I'd forgotten the comment between the time I read it and the time I looked at this code.

This is not my favorite design. It feels a little like patching an abstraction that's not quite right for the job in order to accommodate how its fit exposes a performance penalty. It addresses a symptom rather than the problem.

Per my comment, there's something that doesn't feel quite right to me about `ExperimentRef` right now. But I'd rather not optimize for extra trips to master and then see where it blows up on us. Where it blows up might suggest what the right design should have been in the first place.

Wish I had something more constructive to contribute.

Reply:
discussed offline. agree it's not ideal, but it's the best compromise we can come up with so far.

Reply:
hmm... not a bad idea. i think my issue with this though is that `client` is not structured in any meaningful way right now. so we could dump all the "high-level-ish" methods in there, but it might get really messy. like, we'd have `wait_for_experiment(id)`, `kill_experiment`, `pause_experiment`, etc. it would probably end up being a huge file. 
discussed offline, but we should just consider removing `client` altogether.

Reply:
Maybe same-same, but my own current preference looks like moving `determined.Determined` functionality into `client` and getting rid of `determined`.

Reply:
https://github.com/determined-ai/determined/pull/6853#discussion_r1218540410